mso/resource_mso_schema_template_service_graph.go

```diff
@@ -578,39 +322,11 @@ func resourceMSOSchemaTemplateServiceGraphDelete(d *schema.ResourceData, m inter
 		graphName = tempVar.(string)
 	}
 
-	templatePath := fmt.Sprintf("/templates/%s/serviceGraphs/%s", templateName, graphName)
-	templatePatchStruct := models.NewTemplateServiceGraph("remove", templatePath, nil)
-
-	sitePayload := make([]models.Model, 0, 1)
-
-	sitePayload = append(sitePayload, templatePatchStruct)
-	var siteParams []interface{}
-	if tempVar, ok := d.GetOk("site_nodes"); ok {
-		siteParams = tempVar.([]interface{})
-
-		for _, site := range siteParams {
-			siteMap := site.(map[string]interface{})
-			if siteMap["site_id"] == "" {
-				return fmt.Errorf("site_id is required in site_nodes list")
-			}
-
-			_, ind, err := getSiteServiceGraphCont(
-				cont,
-				schemaId,
-				templateName,
-				siteMap["site_id"].(string),
-				graphName,
-			)
-
-			if err == nil {
-				sitePath := fmt.Sprintf("/sites/%s-%s/serviceGraphs/%d", siteMap["site_id"].(string), templateName, ind)
-				sitePayload = append(sitePayload, models.NewTemplateServiceGraph("remove", sitePath, nil))
-			}
+	path := fmt.Sprintf("/templates/%s/serviceGraphs/%s", templateName, graphName)
 
-		}
-	}
+	templatePatchStruct := models.NewTemplateServiceGraph("remove", path, nil)
 
-	response, err := msoClient.PatchbyID(fmt.Sprintf("/api/v1/schemas/%s", schemaId), sitePayload...)
+	response, err := msoClient.PatchbyID(fmt.Sprintf("/api/v1/schemas/%s", schemaId), templatePatchStruct)
```

Comment:
Why not use the GetRemovePatchPayload() function from go client?

Reply:
made changes.